--- origsrc/ksirk-4.11.5/ksirk/iris/src/irisnet/corelib/netinterface_unix.cpp	2014-01-02 13:35:51.000000000 -0600
+++ src/ksirk-4.11.5/ksirk/iris/src/irisnet/corelib/netinterface_unix.cpp	2014-02-25 14:04:10.000000000 -0600
@@ -35,7 +35,6 @@
 #endif
 #include <unistd.h>
 #include <net/if.h>
-#include <net/route.h>
 #include <netinet/in.h>
 #include <errno.h>
 #include <unistd.h>
--- origsrc/ksirk-4.11.5/ksirk/iris/src/jdns/jdns_p.h	2014-01-02 13:35:51.000000000 -0600
+++ src/ksirk-4.11.5/ksirk/iris/src/jdns/jdns_p.h	2014-02-25 14:04:10.000000000 -0600
@@ -46,6 +46,8 @@
 # define JDNS_OS_SOLARIS
 #elif defined(__APPLE__) && (defined(__GNUC__) || defined(__xlC__) || defined(__xlc__))
 # define JDNS_OS_MAC
+#elif defined(__CYGWIN__)
+# define JDNS_OS_CYGWIN
 #endif
 
 #ifdef JDNS_OS_WIN
--- origsrc/ksirk-4.11.5/ksirk/iris/src/jdns/jdns_sys.c	2014-01-02 13:35:51.000000000 -0600
+++ src/ksirk-4.11.5/ksirk/iris/src/jdns/jdns_sys.c	2014-02-25 14:04:10.000000000 -0600
@@ -702,6 +702,7 @@ static int my_res_init()
 #endif
 
 #if defined(JDNS_OS_MAC) || defined(JDNS_OS_FREEBSD) || \
+defined(JDNS_OS_CYGWIN) || \
 defined(JDNS_OS_NETBSD) || defined(JDNS_OS_OPENBSD) || defined (JDNS_OS_SOLARIS)
 # define USE_EXTEXT
 #endif
--- origsrc/ksirk-4.11.5/ksirk/iris/src/jdns/qjdns_sock.cpp	2014-01-02 13:35:51.000000000 -0600
+++ src/ksirk-4.11.5/ksirk/iris/src/jdns/qjdns_sock.cpp	2014-02-25 14:39:03.855135200 -0600
@@ -62,11 +62,13 @@
 # ifndef IPV6_JOIN_GROUP
 #  define IPV6_JOIN_GROUP 12
 #  define IPV6_MULTICAST_HOPS 10
+#  ifndef Q_OS_CYGWIN
    struct ipv6_mreq
    {
      struct in6_addr ipv6mr_multiaddr;
      unsigned int ipv6mr_interface;
    };
+#  endif
 # endif
 #endif
 
